scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -385,6 +388,11 @@ class FossId(
                 } catch (e: IllegalStateException) {
                     e.showStackTrace()
                     log.error("Package at url=$url and path='$path' cannot be scanned")
+
+                    createdScans.forEach {
+                        log.warn("Deleting previous scan $it.")
+                        deleteScan(it)
```

Comment:
does this delete all created scans, or just failed scans, or is this the same?

Reply:
It deletes all scans created **in this run** i.e. all scans that passed `createScan`, whether they were successful or not.

Reply:
I don't know that much about FossId. Is there a reason to delete all scans, and not keep at least the successful ones?

Reply:
Some further explanation:
In one ORT run, several scans are created, one for each package to scan.

The idea with this change is that either all these scans are successful, either none is created at all (clean slate).
The reasoning being that a ORT run is created regularly e.g. nightly, and we want to have exactly the same amount of scan for each package.

Reply:
That makes sense. Thanks for explaining.

Reply:
Yes, thanks for the clarification. Could this be added as a comment in the code? I think, this atomicity is an important property, which is not really obvious otherwise.

Reply:
Done.